snap/info.go

```diff
@@ -243,6 +243,26 @@ func (app *AppInfo) ServiceSocketFile() string {
 	return filepath.Join(dirs.SnapServicesDir, app.SecurityTag()+".socket")
 }
 
+func infoFromSnapYamlWithSideInfo(meta []byte, si *SideInfo, validate bool) (*Info, error) {
+	info, err := InfoFromSnapYaml(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	if validate {
```

Comment:
I had it that way and then changed my mind, the question is mostly about name, should we check it twice ? (maybe we can easily)